### PR TITLE
Upgrade to Kotlin 1.7.10 and Jetpack Compose 1.2.1

### PR DESCRIPTION
**Pull request by sbilsky**
**Description**
If you updated project which used paparazzi to Kotlin 1.7.X and Compose to 1.2.X it required to update Kotlin and compose in paparazzi.

**Important note:**
Merging this PR will bump everyone using Paparazzi to suddenly use newer Kotlin and Compose.

**What's changed:**
-compose compiler to 1.3.0 because it's required for Kotlin 1.7.10
-compose runtime and ui versions bumped to 1.2.1
-lifecycle-runtime version bumped to 2.5.0
-saved-state version bumped to 1.2.0
-compose material moved to dependencies.gradle
-coroutines to 1.6.4

### CODE REVIEW

**Comment by dniHze** [comment]
I believe you also need to update the tests to use JPC 1.2.0 unbundled compiler.  👀

File: gradle/dependencies.gradle
Comment (TWiStErRob) [inline gradle/dependencies.gradle:9]: matching coroutines?

Comment (TWiStErRob) [inline gradle/dependencies.gradle:9]: Note: Merging this PR will bump everyone using Paparazzi to suddenly use newer Kotlin. So it's not "support Kotlin 1.7.0", but "require Kotlin 1.7.0"

https://github.com/cashapp/paparazzi/blob/549defcd2591b40aebd011ac0a7eecd74c2ea489/paparazzi/paparazzi/build.gradle#L46

Comment (TWiStErRob) [inline gradle/dependencies.gradle:9]: 1.7.10 is [out](https://kotlinlang.org/docs/releases.html#release-details), any reason for not using it?

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt
Comment (TWiStErRob) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:574]: Does this mean anyone using version this PR goes into, will be required to use Compose 1.2.0? (similar to "support" vs "require"). Also based on this it looks like anyone using Compose 1.1.1 cannot use Paparazzi? (and probably that was your motivation with the PR)

File: gradle/dependencies.gradle
Comment (sbilsky) [reply gradle/dependencies.gradle]: yes, but anyway updated it to 1.6.4

Comment (sbilsky) [reply gradle/dependencies.gradle]: Yes,  if you updates the kotlin and compose version in project it will broke paparazzi.
Thanks for comment, I'll update PR title and description to provide more context.

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt
Comment (sbilsky) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:574]: Yes!
My bad, I'll update this info in PR description.
Currently in our project we live with sources builded from this MR in our local maven.

File: gradle/dependencies.gradle
Comment (sbilsky) [reply gradle/dependencies.gradle]: Description and title for PR updated.
Thanks for comment!

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt
Comment (sbilsky) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:574]: Description and title for PR updated.
Thanks for comment!

**Comment by sbilsky** [comment]
@TWiStErRob Thanks for review!  If something else I'll need to update let me know.

File: gradle/dependencies.gradle
Comment (sbilsky) [reply gradle/dependencies.gradle]: When I opened PR, It was blocked by compose compiler version which wasn't stable.
Stable Compiler version was released in 10th August.
https://developer.android.com/jetpack/androidx/releases/compose
https://developer.android.com/jetpack/androidx/releases/compose-kotlin
Currently I updated Kotlin to 1.7.10 and Compose to 1.2.1 with stable compiler version 1.3.0.

**Comment by TWiStErRob** [comment]
I was just curious and wanted to see what's this, I don't have authority here 😅.
Maintainers, I hope my review helped to move this along / prioritize it.

File: gradle/dependencies.gradle
Comment (jrodbx) [inline gradle/dependencies.gradle:28]: nit: move to composeUi.material?

**Review by jrodbx — APPROVED** [review]

File: gradle/dependencies.gradle
Comment (sbilsky) [reply gradle/dependencies.gradle]: Done

**Pull request merged**
